tests/by-util/test_hashsum.rs

```diff
@@ -649,3 +829,75 @@ fn test_check_check_ignore_no_file() {
         .fails()
         .stderr_contains("in.md5: no file was verified");
 }
+
+#[test]
+fn test_check_directory_error() {
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+
+    at.mkdir("d");
+    at.write("in.md5", "d41d8cd98f00b204e9800998ecf8427f  d\n");
+    #[cfg(not(windows))]
+    {
+        scene
+            .ccmd("md5sum")
+            .arg("--check")
+            .arg(at.subdir.join("in.md5"))
+            .fails()
+            .stderr_contains("md5sum: d: Is a directory\n");
+    }
+    #[cfg(windows)]
+    {
+        // Windows has a bit different error message
+        scene
+            .ccmd("md5sum")
+            .arg("--check")
+            .arg(at.subdir.join("in.md5"))
+            .fails()
+            .stderr_contains("md5sum: d: Permission denied\n");
+    }
```

Comment:
With the exception of the expected error messages the two code blocks are identical. And so my suggestion is to move the error messages to a platform-dependent var and make the rest platform-independent and remove the duplicate code. As a side effect, the comment is no longer necessary.